diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -365,6 +411,18 @@ int DifferentialFunctionComparator::cmpOperations(
                 findTypeDifferences(FL, FR, L->getFunction(), R->getFunction());
             }
         }
+        // Handle inverse conditions
+        if (isa<CmpInst>(L) && isa<CmpInst>(R)) {
+            auto CmpL = dyn_cast<CmpInst>(L);
+            auto CmpR = dyn_cast<CmpInst>(R);
+
+            // It is sufficient to compare the predicates here since the
+            // operands are compared in cmpBasicBlocks.
+            if (CmpL->getPredicate() == CmpR->getInversePredicate()) {
+                inverseConditions.emplace(L, R);
+                return 0;
```

Comment:
This seems to compare all `CmpInst`s with the same arguments and inverse predicates as equal, even those that are used at other places than `BranchInst`s. For example these two (clearly semantically different) programs now compare as equal:
```c
#include <stdio.h>

int main() {
    int a = 4, b = 5;
    int cmp = a <= b;
    printf("%d\n", cmp);
}
```
```c
#include <stdio.h>

int main() {
    int a = 4, b = 5;
    int cmp = a > b;
    printf("%d\n", cmp);
}
```
Going through uses of the `CmpInst` and making sure all of them are `BranchInst`s should fix this problem.